### ISSUE TITLE
private/publicSSLKeys not generated

### ISSUE THREAD

**Issue by mhus** [hypery2k/owncloud#338]
Looks like SSL Keys are not generated in every case. The function 'getPublicKey' should generate the keys if not found.

**Comment by mckaygerhard** [comment]
seem same problem in my las log in the isue #338

**Comment by TerraRick** [comment]
I have the same issue. I tracked the problem back to file RoundCubeApp.class.php in function generateKeyPair. Error occurs in line 104 when openssl_pkey_new() function is called. This function needs a proper openssl setup on the machine, which i don't have. I've added an openssl_error_string() call right after the openssl_pkey_new(); and it outputs:

error:02001002:system library:fopen:No such file or directory
error:2006D080:BIO routines:BIO_new_file:no such file
error:0E064002:configuration file routines:CONF_load:system lib
error:02001002:system library:fopen:No such file or directory
error:2006D080:BIO routines:BIO_new_file:no such file","level
error:0E064002:configuration file routines:CONF_load:system lib

Suggestion ... code can be altered to be self-supporting by passing the required arguments in an array to the function (see http://php.net/manual/en/function.openssl-pkey-new.php). 

Hope to hear from you soon ... best regards
